### PR TITLE
damoreno booking-up-for-beauty solution

### PR DESCRIPTION
**Pull request by davidmogar**

### CODE REVIEW

File: booking-up-for-beauty/damoreno/booking_up_for_beauty.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    return hour >= 12 && hour < 18
+}
+
+// Description returns a formatted string of the appointment time
+func Description(date string) string {
+    parsedDate := parseDate("1/2/2006 15:04:05", date)
+
+    return fmt.Sprintf("You have an appointment on %s.",
+                       parsedDate.Format("Monday, January 2, 2006, at 15:04"))
```
Comment (smunilla) [inline booking-up-for-beauty/damoreno/booking_up_for_beauty.go:40]: That's clever!

File: booking-up-for-beauty/damoreno/booking_up_for_beauty.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import (
+    "fmt"
+    "time"
+)
+
+func parseDate(layout, date string) time.Time {
+    parsedDate, err := time.Parse(layout, date)
+
+    if err != nil {
+        fmt.Println("Oops:", err)
```
Comment (hemartin-at-redhat) [inline booking-up-for-beauty/damoreno/booking_up_for_beauty.go:12]: Here I think you should exit. On error you would continue the program, leading to unknown failures down the line. A shame that the tests do not include failures.

File: booking-up-for-beauty/damoreno/booking_up_for_beauty.go (hunk 1 above)
Comment (enarha) [reply booking-up-for-beauty/damoreno/booking_up_for_beauty.go:40]: You can do without fmt.Sprintf here, just concatenate 2 strings. And the final `.` can be part of the layout like `Monday, January 2, 2006, at 15:04.`

File: booking-up-for-beauty/damoreno/booking_up_for_beauty.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import (
+    "fmt"
+    "time"
+)
+
+func parseDate(layout, date string) time.Time {
+    parsedDate, err := time.Parse(layout, date)
+
+    if err != nil {
+	panic("The suplied date is not parseable")
```
Comment (enarha) [inline booking-up-for-beauty/damoreno/booking_up_for_beauty.go:12]: Not sure if the real error (the one `err` is holding) is printed that way, but if it's not you can take the value from err as a string by `err.Error()` and add it to your custom message.

File: booking-up-for-beauty/damoreno/booking_up_for_beauty.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+func Description(date string) string {
+    parsedDate := parseDate("1/2/2006 15:04:05", date)
+
+    return fmt.Sprintf("You have an appointment on %s.",
+                       parsedDate.Format("Monday, January 2, 2006, at 15:04"))
+}
+
+// AnniversaryDate returns a Time with this year's anniversary
+func AnniversaryDate() time.Time {
+    return time.Date(time.Now().Year(), 9, 15, 0, 0, 0, 0, time.UTC)
```
Comment (enarha) [inline booking-up-for-beauty/damoreno/booking_up_for_beauty.go:45]: That makes more sense than what I did, constructing a string and then parsing it. At least the name of the method sounds better, you use Date to create a date :-)

**Review by enarha — APPROVED** [review]

**Review by sudo-happybhati — COMMENTED** [review]
Different way same results very neat solution.
LGTM

**Pull request merged**